lib/page/get-display-value/get-display-value.js

```diff
@@ -41,32 +41,41 @@ const getDisplayValue = (pageInstance, userData, nameInstance) => {
     }
   } else {
     if (nameInstance.items) {
-      const nameItem = nameInstance.items.filter(item => item.value === answer)[0]
+      const nameItem = nameInstance.items.find(({value}) => value === answer)
       answer = nameItem ? getInstanceTitleSummary(nameItem._id) : answer
       if (nameItem) {
         substitution = true
       }
     }
   }
+
   if (_type === 'fileupload') {
     if (Array.isArray(answer)) {
-      answer = answer.map(a => {
-        return `${a.originalname} (${bytes(a.size)})`
-      }).join('\n\n')
+      answer = answer.map(({originalname, size}) => `${originalname} (${bytes(size)})`).join('\n\n')
       markdown = true
     }
   }
 
-  const unansweredQuestion = !answer
-
-  if (unansweredQuestion) {
-    answer = 'Not answered'
-  }
-
   if (Array.isArray(answer)) {
     answer = answer.join('\n\n')
     markdown = true
+  } else {
+    /**
+     *  Answers may have been coerced to a non-string data type so we can't expect
+     *  truthiness to tell us whether the question has been answered. Instead,
+     *  we have some explicit conditions
+     *
+     *  They are:
+     *    - Guard against undefined or null
+     *    - Coerce to a string and confirm that the string is not empty
+     *
+     *  Any other primitive is acceptable
+     */
```

Comment:
Generally not a fan of comments, but I get that there are no abstractions with good names here

Reply:
Yes, it shouldn't be necessary

I write lots of comments in nasty code, barely any comments at all in nice code :)